### PR TITLE
short circuit solver unit test

### PR DESCRIPTION
**Pull request by nitbharambe**
Solver unit test with a source line and fault. Test solid and non solid fault of all types

### CODE REVIEW

**Comment by mgovers** [comment]
since you're rebased off of my branch, you'll have to change base to `feature/short-circuit-solver`

File: tests/cpp_unit_tests/test_math_solver.cpp [hunk 1]
```diff
@@ -536,6 +556,264 @@ TEST_CASE("Test math solver") {
...
+    return sc_output_ref;
+}
+
+// TODO Un-skip unit test later
+TEST_CASE("Short circuit solver" * doctest::skip(true)) {
+    /*
+
+    source -- bus --- line -- bus -- fault(type varying as per subcase)
+
+    */
```
Comment (mgovers) [inline tests/cpp_unit_tests/test_math_solver.cpp:648]: please use `//` for this type of comments

**Review by mgovers — COMMENTED** [review]
a couple small remarks but overall looks good

File: tests/cpp_unit_tests/test_math_solver.cpp [hunk 2]
```diff
@@ -83,6 +87,26 @@ void assert_output(MathOutput<sym> const& output, MathOutput<sym> const& output_
     }
 }
 
+template <bool sym>
```
Comment (nitbharambe) [inline tests/cpp_unit_tests/test_math_solver.cpp:90]: Should this and the existing function for powerflow be in a unnamed namespace too? @mgovers

Comment (mgovers) [reply tests/cpp_unit_tests/test_math_solver.cpp:90]: technically yes, but keep it out of scope for now.

**Comment by mgovers** [comment]
The failing function on `macos` is being fixed by @petersalemink95 on his branch. we do not need to care about that in this PR.

**Review by mgovers — APPROVED** [review]
the code looks really clean and readable now. i can understand exactly what's happening, what's being tested and why something fails or passes.

**Pull request merged**